rasa/core/policies/ensemble.py

```diff
@@ -1,31 +1,81 @@
-import importlib
-import json
+"""Notes.
+
+NOTE: *not* covered here / other changes required
+-  `validate` contains checks that where previously done during intialisation of
+    the ensemble
+    --> trigger in graph validation
+    --> see: `validate` functions in ensembles
+-  `set_rule_data` was done during intialisation and as last step of train
+    --> e.g. persist rule_only_data separately during RulePolicy persist and
+        create a graph component to the prediction graph that loads this specific
+        resource during prediction and has access to all policies and can modify those
+        graph components after they've been loaded, or maybe add a component that
+        after training loads the rule data and adds that resource to all policies
+        by persisting it to each policy resource (?) (FIXME)
+    --> see: `propagate_specification_of_rule_only_data` below
+-  final prediction requires access to domain
+   --> add domain provider in predict graph
+-  final prediction needs to add events to the tracker if and only if the sequence of
+   events seen so far ends with a `UserUttered` event
+   --> add tracker provider to predict graph
+- specific unit tests related to action unlikely intent which basically test whether the
+  *semantic* / idea behind the simple ensemble makes sense
+   --> should replace this with some test in unexpecTEDintentPolicy which test that
+       the probability, priority, ... whatever is chosen such that the tests that were
+       previously here would pass (there is no real need to actually use the ensemble
+       in these tests if no one decides to change the meaning of the simple policy
+       instead of parametrising it's behavior)
+NOTE: differences to previous version
+-  previously there was a comment there saying
+    # find rejected action before running the policies
+    # because some of them might add events
+   this seems to be outdated because the policy prediction contains the event lists
+   and because I could not find policies adding events while they predict
+   --> we do *not* move this to a component that is executed before the policies
+   but leave this snippet in the ensemble's predict
+-  refactored the way the "best prediction" is determined because that decision
```

Comment:
I know it's very tempting to refactor old code as we do these migrations. But I really think now is not the ideal time. We have _a huge_ amount of changes as part of 3.0, and I think our aim should be to minimize unneeded changes as much as possible. The more things we change, the harder it will be to debug if things don't work properly when we start the integration change, remember these new migrated components aren't being tested together yet! 
It will be much easier to manage and test refactors once we have everything up and running.

Reply:
Sorry I should've left the ensembling part itself as un-changed as possible. - But the first part on what is described as change there would mean that if we don't change that we'd need a new component to mimic what was done before.

Reply:
1. Do we want to add a separate component to extract the rejected action before processing the policies, or do we all agree that policies do not alter trackers during their process - @joejuzl  @JEM-Mosig  ? 
2. Do we want to keep the "best prediction computation" as is - or (and this is what it boils down to in the end) replace the two `if` statements by something readable (see comment/explanation below) (because the rest is really just reducing the iterative filtering of lists into a single filter/loop)?  Remember that the tests still need to change anyway because we're not working with policies but predictions. @joejuzl @JEM-Mosig ?

Reply:
I would stick to the bare minimum required to convert to a `GraphComponent`. We can always refactor/improve the code later, but right now we are very time-constrained and in the middle of a huge change.

Reply:
Ok so, @joejuzl @wochinge  regarding...
1. As said before - this will either looks similar but change the logic (because rejection is done after predicting) or not change the logic but require more changes (i.e. a component that rejects actions before the policies predict). @JEM-Mosig  will think about this tomorrow and confirm (as far as I see that comment regarding "this needs to be done before" is obsolete - otherwise the whole "let's put events into policy predictions" wouldn't be there I think)
2. Just reverted the to the old if-logic with necessary adaptions (i.e. list of predictions instead of dictionary of policies). Tests stay as "changed" as they are - because there's no more need to create policies that predict hard-coded constants, since this new component works with predictions. @JEM-Mosig  thinks those changes are ok and will double check those tests that as well.